### PR TITLE
Pilot unit test improvements

### PR DESCRIPTION
**Pull request by howardjohn**
Currently, pilot tests generally spin up a full bootstrap.Server, which has a lot of overhead. This changes to use the minimal fake server, using an in memory buffcon listener

Additionally, there are various fixes for tests - comments inline

Before: 80s
After: 35s (still way too long)

### CODE REVIEW

File: pilot/pkg/bootstrap/configcontroller.go [hunk 1]
```diff
@@ -90,12 +90,6 @@ func (s *Server) initConfigController(args *PilotArgs) error {
 		}
 	}
 
-	// Used for tests.
-	memStore := memory.Make(collections.Pilot)
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/configcontroller.go:94]: dead code

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -222,7 +218,6 @@ func NewServer(args *PilotArgs) (*Server, error) {
 		return nil, err
 	}
 
-	s.initGenerators()
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go:225]: refactored out to a different part of the code for re-use

File: pilot/pkg/bootstrap/server.go [hunk 3]
```diff
@@ -478,6 +473,7 @@ func (s *Server) initDiscoveryService(args *PilotArgs) error {
 	log.Infof("starting discovery service")
 	// Implement EnvoyXdsServer grace shutdown
 	s.addStartFunc(func(stop <-chan struct{}) error {
+		log.Infof("Starting ADS server")
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go:476]: move log so its not triggered in tests. IMO its callers responsibility to log anyways

File: pilot/pkg/xds/discovery.go [hunk 4]
```diff
@@ -165,6 +163,11 @@ func NewDiscoveryServer(env *model.Environment, plugins []string) *DiscoveryServ
 		debugHandlers:           map[string]string{},
 		adsClients:              map[string]*Connection{},
 		serverReady:             false,
+		debounceOptions: debounceOptions{
```
Comment (howardjohn) [inline pilot/pkg/xds/discovery.go:166]: refactor debounce options to not depend on global var

File: pilot/pkg/xds/eds_test.go [hunk 5]
```diff
@@ -111,12 +110,38 @@ func TestEds(t *testing.T) {
...
+		bytes, err := ioutil.ReadFile(filepath.Join(env.IstioSrc, fpath))
+		if err != nil {
+			t.Fatal(err)
+		}
+		result += "---\n"
+		result += string(bytes)
+	}
+	return result
+}
+func mustReadfolder(t *testing.T, folder string) string {
```
Comment (ramaraochavali) [inline pilot/pkg/xds/eds_test.go:125]: nit: new line after function

**Pull request merged**